### PR TITLE
Add mask to sgkit SampleData

### LINKED ISSUE #786: sgkit: Support `call_genotype_mask`?
Often when running  tsinfer, a subset of the variants are wanted for inference. Making a subset sgkit dataset isn't ideal, as it means making a copy of the dataset and rechunking (see https://github.com/pystatgen/sgkit/issues/981).

One option is to have tsinfer support a mask as part of the input dataset. I think I'm right in saying that for tsinfer `variant_mask` would make the most sense - but in sgkit `call_genotype_mask` is standard. We could use `call_genotype_mask` and error if a variant has a mixture of masked and unmasked calls?

The main changes to enable this are in the chunk iterator, which would need to skip the masked rows.

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #786

### CODE REVIEW

File: tsinfer/formats.py [hunk 1]
```diff
@@ -243,21 +243,32 @@ def zarr_summary(array):
...
 
-def chunk_iterator(array, indexes=None):
+def chunk_iterator(array, indexes=None, mask=None):
     """
     Utility to iterate over closely spaced rows in the specified array efficiently
     by accessing one chunk at a time (normally used as an iterator over each row)
     """
+    if mask is not None:
+        assert len(mask) == len(array)
+
```
Comment (jeromekelleher) [inline tsinfer/formats.py:253]: Maybe simpler to create default mask (keep everything)?

**Review by jeromekelleher — COMMENTED** [review]
Looks nice, not particularly invasive!

**Comment by benjeffery** [comment]
This needs a few more tests before merging.

**Comment by benjeffery** [comment]
Added all the other arrays that the existing sample data used, including fallbacks for when they are missing.
Everything that isn't made by sgkit's `vcf_to_zarr` is optional.

**Review by jeromekelleher — APPROVED** [review]
LGTM - looks like we're missing coverage on one error condition, good to catch that before merge

**Pull request merged**